tests/admin_scripts/tests.py

```diff
@@ -1340,15 +1338,9 @@ def test_no_database(self):
 
     def test_readonly_database(self):
         """
-        Ensure runserver.check_migrations doesn't choke when a database is read-only
-        (with possibly no django_migrations table).
+        Ensure runserver.check_migrations doesn't choke when a database is read-only.
         """
-        with mock.patch.object(
-                MigrationRecorder, 'ensure_schema',
-                side_effect=MigrationSchemaMissing()):
-            self.cmd.check_migrations()
-        # Check a warning is emitted
-        self.assertIn("Not checking migrations", self.output.getvalue())
+        self.cmd.check_migrations()
```

Comment:
Is this still simulating a read-only database if the mocking is removed? This test might be obsolete considering the exception catching in `check_migrations()` is removed. Some low level tests for the modified `MigrationRecorder` methods might be in order instead.